tools/rotate-certs.sh

```diff
@@ -0,0 +1,55 @@
+#!/bin/bash
+set -euo pipefail
+
+namespace=openshift-cnv
+_kubectl="${KUBECTL_BINARY:-oc}"
+
+if ! options=$(getopt -o n: -- "$@")
+then
+    exit 1
+fi
+
+eval set -- "$options"
+
+while true; do
+    case "$1" in
+    -n)
+        shift; # The arg is next in position args
+        namespace=$1
+        ;;
+    --)
+        shift
+        break
+        ;;
+    esac
+    shift
+done
+shift $((OPTIND-1))
+
+echo "# Rotating kubemacpool certificates ..."
+${_kubectl} --namespace "${namespace}" delete pods -l app=kubemacpool
+
+echo "# Rotating cdi certificates ..."
+# first rotate the certificates and CAs
+${_kubectl} scale --namespace "${namespace}" --replicas=0 deployment/cdi-operator
+${_kubectl} delete secrets --namespace "${namespace}" -l cdi.kubevirt.io
+# second restart the pods, so that nothing wrong is cached
+${_kubectl} delete pods --namespace "${namespace}" -l cdi.kubevirt.io
+# then delete registrations
+${_kubectl} delete validatingwebhookconfigurations --ignore-not-found=true  --namespace "${namespace}" cdi-api-datavolume-validate
+${_kubectl} delete mutatingwebhookconfigurations --ignore-not-found=true  --namespace "${namespace}" cdi-api-datavolume-mutate
+if [[ "$(${_kubectl} api-resources | grep routes)" =~ "routes" ]]
+then 
+    ${_kubectl} delete routes --ignore-not-found=true --namespace "${namespace}" cdi-uploadproxy
+fi
+# finally restart again, so that all registrations get recreated
+${_kubectl} delete pods --namespace "${namespace}" -l cdi.kubevirt.io
```

Comment:
do this for all namespaces not just the namespace CDI is installed in.  this will get all import/upload/clone pods in progress

Reply:
Done. Also made the route-detection simpler and more robust. If any add-on is not discoverable, `kubectl get api-resources` just returns an error instead of the discovered list.